### PR TITLE
Add output trail to (.txt) file

### PR DESCRIPTION
**Pull request by OannesGPT**
Defines a method to write to file and demonstrates this by appending the generated results to a .txt file 

For easier read and access in post-result analysis or during runtime

Added=
def write_result_to_file(filename, result):

 write_result_to_file("output_trail.txt", result)

### CODE REVIEW

File: babyagi.py [hunk 1]
```diff
@@ -145,6 +145,24 @@ def context_agent(query: str, n: int):
     sorted_results = sorted(results.matches, key=lambda x: x.score, reverse=True)
     return [(str(item.metadata['task'])) for item in sorted_results]
 
+# OannesGPT Additions - Function to write result to file and append on top of existing file
```
Comment (davidbyttow) [inline babyagi.py:148]: Please delete these comments, unnecessary

File: babyagi.py [hunk 2]
```diff
@@ -145,6 +145,24 @@ def context_agent(query: str, n: int):
...
+        existing_data = ''
+
+    # Combine the existing data and the new variable
+    new_data = result + '\n\n' + existing_data
+
+    # Write the new data to the output file
+    with open(filename, 'w') as f:
+        f.write(new_data)
+
+# End of: Oannes Additions
```
Comment (davidbyttow) [inline babyagi.py:164]: delete

File: babyagi.py [hunk 3]
```diff
@@ -171,6 +189,9 @@ def context_agent(query: str, n: int):
         this_task_id = int(task["task_id"])
         print("\033[93m\033[1m"+"\n*****TASK RESULT*****\n"+"\033[0m\033[0m")
         print(result)
+        
+        # Step OannesGPT Addition  - Attempt to write output to .txt file
```
Comment (davidbyttow) [inline babyagi.py:193]: delete

File: babyagi.py [hunk 4]
```diff
@@ -171,6 +189,9 @@ def context_agent(query: str, n: int):
...
+        write_result_to_file("output_trail.txt", result)
```
Comment (davidbyttow) [inline babyagi.py:194]: Is this better than running and outputting to file? E.g., `python babyagi.py > output_trail.txt`?

if so, this should only be done if there is a commandline arg to do it. 

Can you say more about the use case?

**Review by davidbyttow — CHANGES_REQUESTED** [review]

**Review by DrewMcArthur — COMMENTED** [review]
what benefit, if any, does this have over just

`python main.py >> output.txt`, which doesn't require any changes?

File: babyagi.py (hunk 1 above)
Comment (OannesGPT) [reply babyagi.py]: c

File: babyagi.py (hunk 2 above)
Comment (OannesGPT) [reply babyagi.py]: @davidbyttow Hey, I updated the file without notes ;)Sorry am new to this, wanted to be clear

File: babyagi.py (hunk 4 above)
Comment (OannesGPT) [reply babyagi.py:192]: Use case is to properly study how well data is being remembered and processed by the bot. I use it to generate python scripts and I want to see if he adds for example every script he manages to write he put up in his tasklist

**Pull request closed without merging**